iet.so/src/iet_worker.cpp

```diff
@@ -242,49 +242,59 @@ bool IETWorker::do_iet_power_stress(void) {
+            std::to_string(gpu_id) + " " + " Total time in ms " + " " + std::to_string(total_time_ms) +
+            " Run duration in ms " + " " + std::to_string(run_duration_ms);
         rvs::lp::Log(msg, rvs::logtrace);
 
         if (total_time_ms > run_duration_ms) {
             break;
-	}
+        }
 
-       //It doesnt make sense to read power continously so slowing down
-       sleep(1000);
+        //It doesnt make sense to read power continously so slowing down
+        sleep(1000);
 
-       // check if stop signal was received
-       if (rvs::lp::Stopping())
-         return true;
-       }
-       // json log the avg power
-       log_to_json(IET_AVERAGE_POWER_KEY, std::to_string(max_power),
-                rvs::loginfo);
-       if(max_power >= target_power) {
-             msg = "[" + action_name + "] " + MODULE_NAME + " " +
-                     std::to_string(gpu_id) + " " + " Average power met the target power :" + " " + std::to_string(max_power);
-            rvs::lp::Log(msg, rvs::loginfo);
-            result = true;
-       }else{
-            msg = "[" + action_name + "] " + MODULE_NAME + " " +
-                     std::to_string(gpu_id) + " " + " Average power couldnt meet the target power  \
-                     in the given interval, increase the duration and try again, \
-                     Average power is :" + " " + std::to_string(cur_power_value);
-            rvs::lp::Log(msg, rvs::loginfo);
-            result = false;
-       }
-
-       msg = "[" + action_name + "] " + MODULE_NAME + " " +
-                   std::to_string(gpu_id) + " " + " End of worker thread " ;
-       rvs::lp::Log(msg, rvs::loginfo);
-
-       endtest = true;
-
-       return result;
+        // check if stop signal was received
+        if (rvs::lp::Stopping())
+            return true;
+    }
+
+    // json log the avg power
+    log_to_json(IET_AVERAGE_POWER_KEY, std::to_string(max_power),
+            rvs::loginfo);
+    if(max_power >= target_power) {
+        msg = "[" + action_name + "] " + MODULE_NAME + " " +
+            std::to_string(gpu_id) + " " + " Average power met the target power :" + " " + std::to_string(max_power);
+        rvs::lp::Log(msg, rvs::loginfo);
+        result = true;
+    }else{
+        msg = "[" + action_name + "] " + MODULE_NAME + " " +
+            std::to_string(gpu_id) + " " + " Average power couldnt meet the target power  \
+            in the given interval, increase the duration and try again, \
+            Average power is :" + " " + std::to_string(cur_power_value);
+        rvs::lp::Log(msg, rvs::loginfo);
+        result = false;
+    }
+
+    msg = "[" + action_name + "] " + MODULE_NAME + " " +
+        std::to_string(gpu_id) + " " + " End of worker thread " ;
+    rvs::lp::Log(msg, rvs::loginfo);
+
+    endtest = true;
+
+    if (true == t.joinable()) {
+
+        try {
+            t.join();
+        }
+        catch (std::exception& e) {
```

Comment:
if thread is joinable, do we still check for exceptions?

Reply:
This just for debug purpose just in case join() call fails ?